### PR TITLE
Signin redirect not working #2753

### LINKED ISSUE #2753: Signin redirect not working
`keystone.set('signin redirect', '/redirect-url');` used to work, but isn't working anymore.

Looks like it's missing from the new `createDynamicRouter.js` (https://github.com/keystonejs/keystone/blob/master/admin/server/app/createDynamicRouter.js)

Signin redirect option in [Keystone Doc](http://keystonejs.com/docs/configuration/#options-database).

@mxstbr Could you have a look and confirm if it's the case?

Thanks!

### PR DESCRIPTION
**Pull request by jdr0dn3y**
Fix #2753
@mxstbr 

Checks to see if "signin redirect" is anything other than "signin" and if it is, redirects to that page, otherwise, uses default actions.

### CODE REVIEW

File: admin/server/routes/signin.js [hunk 1]
```diff
@@ -1,9 +1,12 @@
 var keystone = require('../../../');
-
-module.exports = function (req, res) {
-	keystone.render(req, res, 'signin', {
-		submitted: req.body,
-		from: req.query.from,
-		logo: keystone.get('signin logo'),
-	});
+	var signin = keystone.get('signin redirect') || 'signin';
+	if (signin != 'signin') {
```
Comment (mxstbr) [inline admin/server/routes/signin.js]: nit: strict equality check `!==`

**Comment by mxstbr** [comment]
This looks very good, would you mind fixing the Travis error (linting) and the one tiny nitpick? Thanks!

**Comment by jdr0dn3y** [comment]
Updated.

**Comment by mxstbr** [comment]
Still getting linting errors:

```
/home/travis/build/keystonejs/keystone/admin/server/routes/signin.js
  4:2   error  'res' is not defined  no-undef
  6:18  error  'req' is not defined  no-undef
  6:23  error  'res' is not defined  no-undef
  7:14  error  'req' is not defined  no-undef
  8:9   error  'req' is not defined  no-undef
```

**Comment by jdr0dn3y** [comment]
Somehow, I managed to remove ```modules.export``` which is where the extra ```}``` came from in the previous error...

**Comment by jdr0dn3y** [comment]
It's still getting errors, but they are field errors, nothing to do with the signin system.

**Comment by mxstbr** [comment]
Perfect, those are expected! LGTM, thanks so much!

**Comment by jdr0dn3y** [comment]
No problem!

**Comment by JedWatson** [comment]
Actually, `signin redirect` is supposed to redirect you _after_ you sign in - so you can use keystone's signin framework then be directed somewhere custom, e.g. `/dashboard` rather than to the Admin UI. This redirects _from_ the signin page to somewhere else.

We'll need to make sure that variable is exposed to the client side of the Signin page, then take it into account after a successful signin...

**Comment by JedWatson** [comment]
More specifically, we need to override the logic here when `signin redirect` is set:

https://github.com/keystonejs/keystone/blob/master/admin/client/Signin/SigninView.js#L63

**Comment by jdr0dn3y** [comment]
I think I have it this time. Thanks for the pointer @JedWatson

File: admin/client/Signin/SigninView.js [hunk 2]
```diff
@@ -60,7 +60,11 @@ var SigninView = React.createClass({
 				return this.displayError('The email and password you entered are not valid.');
 			} else {
 				// Redirect to where we came from or to the default admin path
-				top.location.href = this.props.from ? this.props.from : Keystone.adminPath;
+				if (Keystone.redirect !== '') {
+					top.location = Keystone.redirect;
```
Comment (JedWatson) [inline admin/client/Signin/SigninView.js]: this should be top.location.href, no?

File: admin/client/Signin/SigninView.js [hunk 3]
```diff
@@ -60,7 +60,11 @@ var SigninView = React.createClass({
 				return this.displayError('The email and password you entered are not valid.');
 			} else {
 				// Redirect to where we came from or to the default admin path
-				top.location.href = this.props.from ? this.props.from : Keystone.adminPath;
+				if (Keystone.redirect !== '') {
```
Comment (JedWatson) [inline admin/client/Signin/SigninView.js]: IMO clearer to do `if (Keystone.redirect)` here - empty strings are falsy

**Comment by jdr0dn3y** [comment]
Passed all relevant checks @mxstbr :-) I have it live in production as well on https://rise.vision

File: admin/client/Signin/SigninView.js (hunk 2 above)
Comment (jdr0dn3y) [inline admin/client/Signin/SigninView.js]: top.location achieves the same thing. I'll change it though to make it more consistent.

File: admin/client/Signin/SigninView.js (hunk 3 above)
Comment (jdr0dn3y) [inline admin/client/Signin/SigninView.js]: Will change it for code quality.

**Comment by JedWatson** [comment]
Perfect! Thanks @jdr0dn3y

**Comment by mxstbr** [comment]
Awesome @jdr0dn3y!!

**Pull request merged**